main/src/com/google/refine/expr/functions/ToString.java

```diff
@@ -54,7 +55,7 @@ public Object call(Properties bindings, Object[] args) {
                     OffsetDateTime odt = (OffsetDateTime)o1;
                     return odt.format(DateTimeFormatter.ofPattern((String)o2));
                 } else if (o1 instanceof Number) {
-                    return String.format((String) o2, (Number) o1);
+                    return new Formatter().format((String) o2, o1).toString();
```

Comment:
From the issue discussion it is not clear to me what difference it makes to use `Formatter::format()` rather than `String.format`. Could you demonstrate the use cases that motivate the change with unit tests?

Reply:
This is actually true. Initially, I thought the Formatter was a better option to use that String.format(), but I did more digging and landed on this [stackoverflow page](https://stackoverflow.com/questions/2809633/what-is-the-difference-between-messageformat-format-and-string-format-in-jdk-1-5#:~:text=String.format%20is%20just%20a%20shortcut%20to%20Formatter%2C%20this,formatting%20convention%2C%20as%20described%20in%20the%20linked%20documentation.) - referring to the last answer on it.

This is how the method is implemented in java too:

```java
public static String format(String format, Object... args) {
        return new Formatter().format(format, args).toString();
}
```

That said, I think I may have misunderstood the issue and will return to the issue's page to clarify all doubts.